src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/MetricsEventSource.cs

```diff
@@ -0,0 +1,356 @@
+        // lambdas can't. Putting all the command handling logic in this nested class
+        // is a simpler way to opt everything out in bulk.
+        private class CommandHandler
+        {
+            private AggregationManager? _aggregationManager;
+            private string _sessionId = "";
+
+            public void OnEventCommand(EventCommandEventArgs command)
+            {
+#if OS_SUPPORT_ATTRTIBUTES
+                if (OperatingSystem.IsBrowser())
+                {
+                    // AggregationManager uses a dedicated thread to avoid losing data for apps experiencing threadpool starvation
+                    // and browser doesn't support Thread.Start()
+                    //
+                    // This limitation shouldn't really matter because browser also doesn't support out-of-proc EventSource communication
+                    // which is the intended scenario for this EventSource. If it matters in the future AggregationManager can be
+                    // modified to have some other fallback path that works for browser.
+                    Logger.Message("System.Diagnostics.Metrics EventSource not supported on browser");
+                    return;
+                }
+#endif
+                if (command.Command == EventCommand.Update || command.Command == EventCommand.Disable ||
+                    command.Command == EventCommand.Enable)
+                {
+                    if (_aggregationManager != null)
+                    {
+                        _aggregationManager.Dispose();
+                        _aggregationManager = null;
+                        Logger.Message($"Previous session with id {_sessionId} is stopped");
+                    }
+                    _sessionId = "";
+                }
+                if ((command.Command == EventCommand.Update || command.Command == EventCommand.Enable) &&
+                    command.Arguments != null)
+                {
+                    if (command.Arguments!.TryGetValue("SessionId", out string? id))
+                    {
+                        _sessionId = id!;
+                        Logger.Message($"SessionId argument received: {_sessionId}");
+                    }
+                    else
+                    {
+                        _sessionId = System.Guid.NewGuid().ToString();
+                        Logger.Message($"New session started. SessionId auto-generated: {_sessionId}");
+                    }
+
+
+                    double defaultIntervalSecs = 1;
+                    double refreshIntervalSecs = defaultIntervalSecs;
+                    if (command.Arguments!.TryGetValue("RefreshInterval", out string? refreshInterval))
+                    {
+                        Logger.Message("RefreshInterval filter argument received: " + refreshInterval);
+                        if (!double.TryParse(refreshInterval, out refreshIntervalSecs))
```

Comment:
>refreshIntervalSecs [](http://example.com/codeflow?start=66&length=19)

should we check if this value less than MinCollectionTime?

Reply:
I added an assert that should start firing if someone accidentally makes default < MinCollectionTime. Is that what you meant here?

Reply:
yes. I am just wondering instead of the assert you rest the value to the min value?

Reply:
If a dev changes either the minimum or the default and it causes the default to no longer be valid I'd like them to see their mistake quickly and address it. I worry that coercing the value would obscure the issue and leave the dev unaware that their change had unintended results.